docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,432 @@
+
+We could have directly used the remote device to communicate with the metadata server,
+but this approach (similar to the one used to create devices from a recovery device)
+has the advantage of allowing to provide a list of linked browsers/computers.
+
+For existing devices (created before/without the authentication service) the client would need to create and upload a remote device to the service (cf [Uploading a new device](#uploading-a-new-device)) in order to access Parsec from a web environment.
+
+
+### Global flow diagram
+
+```mermaid
+flowchart TD
+  H1{PBKDF_1}
+  H2{PBKDF_2}
+
+  Email & Password --> H1 & H2
+
+  H1 --> AUTH_MEDIUM_HMAC_KEY
+
+  H2 --> AUTH_MEDIUM_SYM_KEY_PARAMS & AUTH_MEDIUM_SYM_KEY
+
+  CSPRNG(Cryptographic Pseudo-random Generator)
+
+  CSPRNG --> AUTH_MEDIUM_PRIV_KEY --> AUTH_MEDIUM_PUB_KEY
+  CSPRNG --> ACCOUNT_MANIFEST_SYM_KEY
+
+  AUTH_MEDIUM_SYM_KEY & AUTH_MEDIUM_PRIV_KEY --> E1{"Encrypt the private key"} --> enc_PRIV_KEY["enc(AUTH_MEDIUM_PRIV_KEY)"]
+
+  AUTH_MEDIUM_PUB_KEY --> |"Store in manifest"| ACCOUNT_MANIFEST
+
+  ACCOUNT_MANIFEST & ACCOUNT_MANIFEST_SYM_KEY --> E2{"Encrypt account manifest"} --> enc_MANIFEST["enc(ACCOUNT_MANIFEST)"]
+
+  AUTH_MEDIUM_PUB_KEY & ACCOUNT_MANIFEST_SYM_KEY --> E3{"Encrypt account manifest symmetric key"} --> enc_MANIFEST_KEY["enc(ACCOUNT_MANIFEST_SYM_KEY)"]
+
+  CSPRNG --> DEVICE_SYM_KEY
+
+  DEVICE_SYM_KEY & DEVICE --> E4{"Encrypt the device"} --> enc_DEVICE["enc(DEVICE)"]
+
+  DEVICE_SYM_KEY --> |"Store in manifest"| ACCOUNT_MANIFEST
+
+  subgraph Parsec-Auth server
+    AUTH_MEDIUM_HMAC_KEY
+    AUTH_MEDIUM_SYM_KEY_PARAMS
+    enc_PRIV_KEY
+    enc_MANIFEST
+    enc_MANIFEST_KEY
+    enc_DEVICE
+  end
+
+  E[Email] & P[Password] & AUTH_MEDIUM_SYM_KEY_PARAMS --> H3{PBKDF_2} --> AUTH_SYM_KEY[AUTH_MEDIUM_SYM_KEY]
+
+  AUTH_SYM_KEY & enc_PRIV_KEY --> D1{"Decrypt the private key"} --> PRIV_KEY[AUTH_MEDIUM_PRIV_KEY]
+
+  PRIV_KEY & enc_MANIFEST_KEY --> D2{"Decrypt the account manifest symmetric key"} --> MANIFEST_KEY[ACCOUNT_MANIFEST_SYM_KEY]
+
+  MANIFEST_KEY & enc_MANIFEST --> D3{"Decrypt the account manifest"} --> MANIFEST[ACCOUNT_MANIFEST]
+
+  MANIFEST --> DEV_KEY[DEVICE_SYM_KEY]
+
+  DEV_KEY & enc_DEVICE --> DEV[DEVICE]
+```
+
+## Alternatives Considered
+
+N/A
+
+## Operations
+
+That require managing a new service that will store the devices of the user.
+
+## Security/Privacy/Compliance
+
+Consideration should be taken about the method used to save the device on the remote server.
+The service would need to identify the user so we will need to have some information about the user (email mostly).
+
+### Consideration on PBKDF algorithms
+
+The `PBKDF_A` (used to derive a secret from the password) and `PBKDF_B` (used to derive the private key from the password) output should be different in order to avoid leaking the private key.
+
+To make the output different, we could:
```

Comment:
Keep things simple: we already use argon2, so that's what we are going to use, with the salt composed of the email + a constant unique for each derivation.

Reply:
The goal was to note the different possibilities.
I've added our choice for using the salt method at the end of the list